### PR TITLE
User routes working

### PR DESCRIPTION
**Pull request by IqbalAshif**

### CODE REVIEW

File: services/garage-backend/package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "@types/jsonwebtoken": "^8.5.5",
     "@types/multer": "^1.4.7",
     "@types/node": "^16.11.6",
+    "@types/supertest": "^2.0.11",
```
Comment (serushakov) [inline services/garage-backend/package.json:20]: what is this for?

File: services/garage-backend/src/services/userService.ts [hunk 2]
```diff
@@ -50,3 +52,48 @@ export const upsertUser = async (
...
+    });
+}
+
+function deleteUser(userId: string, removedBy: ObjectId, removedAt: Date): Promise<UserDocument> {
+  const update: Partial<UserDocument> = { removedAt, removedBy };
+  return User.findByIdAndUpdate(userId, update, { new: true })
+    .exec()
+    .then((user) => {
+      if (!user) {
+        throw new Error(`Thing ${userId} not found`);
```
Comment (serushakov) [inline services/garage-backend/src/services/userService.ts:88]:
```suggestion
        throw new Error(`User ${userId} not found`);
```

File: services/garage-backend/src/services/userService.ts [hunk 3]
```diff
@@ -50,3 +52,48 @@ export const upsertUser = async (
...
+    .then((user) => {
+      if (!user) {
+        throw new Error(`User ${userId} not found`);
+      }
+      return user;
+    });
+}
+
+function findAllUser() {
+  return User.find().sort({ fullName: 1, email: -1 }).exec();
```
Comment (serushakov) [inline services/garage-backend/src/services/userService.ts:68]:
```suggestion
  return User.find().exec();
```
sorting breaks our deployed versions

**Review by serushakov — COMMENTED** [review]
looks good, some small fixes needed

File: services/garage-backend/package.json (hunk 1 above)
Comment (IqbalAshif) [reply services/garage-backend/package.json:20]: ah this was for testing.

**Review by serushakov — APPROVED** [review]
lgtm

**Pull request merged**